### PR TITLE
Add try to serviceinfo to avoid Nil crash

### LINKED ISSUE #5786: [production][package#show] undefined method `[]' for nil:NilClass
package#show threw this 1 times
<pre>
undefined method `[]&#39; for nil:NilClass
</pre>
[https://build.opensuse.org/package/show/home:duwe:branches:Kernel:HEAD/kernel-default](https://build.opensuse.org/package/show/home:duwe:branches:Kernel:HEAD/kernel-default)"

[See this exception on Errbit](https://errbit-opensuse.herokuapp.com/apps/5620ca2bdc71fa00b1000000/problems/5b8f8beefe4eac0014218d70)

### PR DESCRIPTION
**Pull request by ChrisBr**
Partially revert 0a36736e04123fdbf70f0864fff7ea0b4d103ea2.
Fix #5786.

### CODE REVIEW

File: src/api/app/models/package.rb [hunk 1]
```diff
@@ -947,7 +947,7 @@ def jobhistory_list(project, repository, arch, options = {})
   end
 
   def service_error(revision = nil)
-    revision ||= serviceinfo['xsrcmd5']
+    revision ||= serviceinfo.try { to_hash['xsrcmd5'] }
```
Comment (mdeniz) [inline src/api/app/models/package.rb]: I don't think this is the best solution. I think that the `serviceinfo` method must return a `{}` if there is no `dir['serviceinfo]` is `nil`.

I mean, `serviceinfo` returns a `{}` if there is an exception, so I guess that the default behaviour is that.

Comment (coolo) [reply src/api/app/models/package.rb]: that was the idea yes - so ``dir['serviceinfo'] || {}``

Comment (ChrisBr) [reply src/api/app/models/package.rb]: Yo, for sure not! This is

1) for documentation of the monkey patch
2) I would nevertheless propose to merge it to have a deployable master

Comment (ChrisBr) [reply src/api/app/models/package.rb]: Btw: @mdeniz @saraycp 

This is the output of the Directory.hashed

```
X({"name"=>"kernel-default", "rev"=>"2", "vrev"=>"2", "srcmd5"=>"aae8998dace7a7eb6087d1e362925326", "linkinfo"=>X({"project"=>"home:duwe:branches:Kernel:HEAD", "package"=>"kernel-source", "error"=>"home:duwe:branches:Kernel:HEAD/kernel-source: conflict in file config.tar.bz2"}), "entry"=>X({"name"=>"_link", "md5"=>"581cdadd84890e029027bdc9cde9b707", "size"=>"46", "mtime"=>"1250198147"})})
```

So we should check for error here and return empty hash then?

Comment (saraycp) [reply src/api/app/models/package.rb]: That error is part of the `linkinfo` key. Maybe it is better not to rely on an error inside 'linkinfo` but instead focus on the `serviceinfo` key.

Comment (ChrisBr) [reply src/api/app/models/package.rb]: ah you're right @saraycp ... nevermind!

**Comment by ChrisBr** [comment]
Superseded by https://github.com/openSUSE/open-build-service/pull/5788

**Pull request closed without merging**